services/api/src/resources/env-variables/resolvers.ts

```diff
@@ -38,7 +30,6 @@ const getEnvVarsByProjectId = async (
   if(args.withGroupEnvVars){
     const projectGroups = await models.GroupModel.loadGroupsByProjectId(pid);
```

Comment:
This will allow all users to see all env vars for all groups for a project, regardless of a member is part of a group or not. We either need to rewrite this section so that is uses the same logic as here https://github.com/amazeeio/lagoon/blob/master/services/api/src/resources/group/resolvers.ts#L63

OR, better IMO is we don't surface the group level env vars in the project. Instead, i've added `envVariables` to the group and they can be retrieved like

```graphql
project:projectByName(name: "credentialstest-project1"){
    id
    name
    groups {
      id
      name
      ... on Group {
        envVariables {
          id
          name
          scope
          value
        }
      }
    }
  }
```

since the groups are all added in openshiftbuilddeploy anyway, I don't think it's too much of a problem to do an extra query, or modify the current one to get them via groups